### ISSUE TITLE
Participation in the Package Maintenence WG

### ISSUE THREAD

**Issue by wesleytodd** [expressjs/discussions#77]
I mentioned this in the last TC meeting, but I volunteered to help get the Package Maintenance WG some input and help getting off the ground by being a contact person between the Express ecosystem and what they want to help with.

If you are interested in this work, you can see whats up here: https://github.com/nodejs/package-maintenance

The first step is to fill in a survey.  I will put together full answers to these questions, but before I do I wanted to give everyone else the opportunity to give preliminary input so I don't go too far off from what everyone thinks.

So, here are the questions which will help the WG decide what to do moving forward:

+ What are the biggest problems your package need to face?
  - With a long project history and a strong focus on feature support, we have struggled to make new feature updates while also keeping up with all the other things (docs, support requests, security updates, etc).
+ What are the most time-consuming tasks?
  - Support and issue triage take the most time.  As an important entry point for the whole node.js ecosystem we get a fair bit of simple issues, often not even Express specific issues.  
+ Which are the most difficult thing to implement in your package ecosystem? (ex: CI/CD, Versioned API Docs, etc...)
  - Automated release process.  As it stands there is one release manager (@dougwilson).  He handles it well, but because of long existing security concerns there has been no great way to expand the circle to others.  He has some ideas (maybe willing to expand on them here?) which would manages some of the security concerns as well as making it possible for others to release.
+ What do you think that is missing in your package ecosystem? (ex: CITGM, etc..)
  - CITGM for sure.  Would have helped greatly with a core change which caused a bug in `on-finished`.
  - Issue templates across all the repos
  - More contributors 
+ What are the tasks that don't let you focus in the business code?
  - In the risk of being repetitive, user support requests take the most non-coding time.
+ How is designed you process to add new maintainers in your package?
  - https://github.com/expressjs/express/blob/master/Contributing.md
+ What type of automation could help you?
  - Release process automation with a focus on security. Also, automation on triaging and pointing new contributors to the most important tasks to work on.
+ What are the most frequent problems?
  - Slow progress on new features.
+ What would you like to do to us?
  - Helping point new contributors to the right areas, standardize 
+ What problems are due to your dependencies?
  - False or overblown security reports have caused multiple incidents. We pin our external dependencies which has the negative repercussion of more work to keep updates rolling out.  We do this because it prevents accidental security incidents from a permissive version range.
+ Are your dependencies need of help? 
  - Express is made up of many modules, most of which are directly supported by the Express core team.  So yes they also need help with recruiting new contributors and issue triage help.
+ Do you have to contribute to your dependencies? Do you establish a good communication channel?
  - Almost the entire dependency tree is directly maintained by the same people, so yes.  This is probably uncommon, but it helps keep things secure and bug free.
  - The external deps are all installed with strict versions locking, but when issues arise it has not been a problem to get deps updated (@dougwilson maybe you can give more input here?).  

After a discussion with @mhdawson, we identified a few key things the @expressjs/express-tc can do to help:

1. Identify 3 important key objective where we need champions.  Examples: http2, promise support in middleware, documentation.
2. Identify 10 issues which are the highest priority for the project (should probably align with the 3 main goals)
3. Identify "good first issues" and/or issues where new users can start getting involved easily

This will help the WG give direction to all of the people who want to help.

**Comment by dougwilson** [comment]
Thanks for posting this. I can help out as well. For the iteration, I was thinking we could just edit the OP as we go. Github does show edit history and diffs, so I figure that should probably be good enough, yea?

**Comment by mhdawson** [comment]
Any progress or thoughts about these questions? We were discussing in the latest package maintenance meeting and it would be great to have input from the express ecosystem.

**Comment by wesleytodd** [comment]
Hey, I am sorry for the delay, but I will be checking back in soon and this is my first priority.  In fact I will start on the answers to this today and hopefully have the draft I said I would have by tomorrow.

**Comment by Eomm** [comment]
Hi @wesleytodd, may I ask you some updates?

**Comment by wesleytodd** [comment]
Hey @Eomm & @mhdawson, I updated most of the answers and added a few notes from the conversation I had with @mhdawson.  

Before we submit them, I would like to get @dougwilson's opinion, and maybe we can go over it in a TC meeting.  @dougwilson if we can schedule a TC meeting sometime soon we have enough to discuss.  Would some time next week work for you?

**Comment by dougwilson** [comment]
Hey @wesleytodd thank you so much for the work you put into attending the meetings and updating the original post in this issue. I haven't fully read through all the changes yet, but wanted to respond to your meeting request. I assume by "next week" you mean the week of 6/24, right? Or of 7/1?

I made the placeholder for the meeting at https://github.com/expressjs/discussions/issues/81 . If you meant 7/1 week, we can change the date.

**Comment by wesleytodd** [comment]
Thanks! Yes I did mean this week, the time in #81 works for me.  I will add some items, including this to the agenda.

**Comment by wesleytodd** [comment]
As an update, we discussed this in the last TC meeting and it is mostly good.  Just waiting on a once over from @dougwilson to make sure we didn't miss anything, then I will submit the PR on the WG repo.

**Comment by Eomm** [comment]
@wesleytodd May I open a PR with these answers and discuss those there?

**Comment by wesleytodd** [comment]
@eomm, I see that you opened a scaffold.  We had discussed this in the last TC meeting, but with the US holiday and getting sick after traveling I have not been able to follow up.  I believe we are all on the same page from the express side, so I see no reason not to submit these.

The update in your PR is actually as a part of a separate issue, https://github.com/nodejs/package-maintenance/pull/220, not sure if this is intentional, but I am guessing it would be best to open it as a separate one.  So I will do that now.

**Comment by wesleytodd** [comment]
Answers submitted: https://github.com/nodejs/package-maintenance/pull/230
